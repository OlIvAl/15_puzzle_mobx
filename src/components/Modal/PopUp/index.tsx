import React from 'react';
import {StyledPopUpCloseButton, StyledPopUpContent, StyledPopUpHeader, StyledPopUpWrapper} from './StyledComponents';

interface IProps {
  onClose: () => void;
}

const PopUp: React.FC<IProps> = ({onClose, children}): JSX.Element => (
  <StyledPopUpWrapper>
    <StyledPopUpHeader>
      You WIN!!!
    </StyledPopUpHeader>
    <StyledPopUpContent>
      {children}
    </StyledPopUpContent>
    <StyledPopUpCloseButton
      onClick={onClose}
      type='button'
    >
      Close
    </StyledPopUpCloseButton>
  </StyledPopUpWrapper>
);

export default PopUp;