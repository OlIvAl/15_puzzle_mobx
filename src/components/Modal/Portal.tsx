import React, {CSSProperties} from 'react';
import ReactDOM from 'react-dom';

const modalRoot = document.getElementById('modal-root');

interface IProps {

}

export default
class Portal extends React.Component<IProps> {
  el: HTMLDivElement;

  constructor(props: IProps) {
    super(props);

    this.el = document.createElement('div');
    this.el.style.position = 'fixed';
    this.el.style.top = '0';
    this.el.style.left = '0';
    this.el.style.width = '100%';
    this.el.style.height = '100vh';
    this.el.style.display = 'flex';
    this.el.style.justifyContent = 'center';
    this.el.style.alignItems = 'center';
  }

  componentDidMount() {
    (modalRoot as HTMLDivElement).appendChild(this.el);
  }

  componentWillUnmount() {
    (modalRoot as HTMLDivElement).removeChild(this.el);
  }

  render() {
    return ReactDOM.createPortal(
      this.props.children,
      this.el,
    );
  }
}