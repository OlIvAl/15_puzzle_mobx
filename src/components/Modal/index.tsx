import React from 'react';
import Portal from './Portal';
import PopUp from './PopUp';
import Backdrop from './Backdrop';

interface IProps {
  onClose: () => void;
}

const Modal: React.FC<IProps> = ({onClose, children}): JSX.Element => (
  <Portal>
    <PopUp
      onClose={onClose}
    >
      {children}
    </PopUp>
    <Backdrop
      onClick={onClose}
    />
  </Portal>
);

export default Modal;