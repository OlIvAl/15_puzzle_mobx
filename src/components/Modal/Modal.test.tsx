import * as React from 'react';
import Modal from '.';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

interface IProps {
  onClose: () => void;
}

describe('Modal', () => {
  const onClose = jest.fn();
  const CONTENT: string = 'test content';

  it('он отображается', () => {

    const wrapper = shallow<React.FC<IProps>>(
      <Modal
        onClose={onClose}
      >
        {CONTENT}
      </Modal>
    );

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('ссодержит, переданный в него, контент', () => {
    const wrapper = shallow<IProps>(
      <Modal
        onClose={onClose}
      >
        {CONTENT}
      </Modal>
    );

    expect(wrapper.contains(CONTENT)).toEqual(true);
  });

  it('дочерний компонент PopUp содержит onClose', () => {
    const wrapper = shallow<IProps>(
      <Modal
        onClose={onClose}
      >
        {CONTENT}
      </Modal>
    );

    expect(wrapper.find('PopUp').prop('onClose')).toEqual(onClose);
  });
});