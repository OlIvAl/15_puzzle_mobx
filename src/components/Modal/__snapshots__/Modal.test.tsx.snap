// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal он отображается 1`] = `
<Portal>
  <PopUp
    onClose={[MockFunction]}
  >
    test content
  </PopUp>
  <styled.div
    onClick={[MockFunction]}
  />
</Portal>
`;
